### PR TITLE
Listen: Reconnect if connection is interrupted

### PR DESCRIPTION
**Pull request by svix-frank**
Listen now will automatically re-establish a connection if interupted (like if a deploy causes the websocket to close)

### CODE REVIEW

File: relay/relay.go [hunk 1]
```diff
@@ -199,8 +215,9 @@ func (c *Client) recLoop() {
 		_, packet, err := c.conn.ReadMessage()
 		if err != nil {
 			select {
-			case <-c.done:
-				continue
+			case <-c.stopRead:
+				// dont send error if
+				// stop was requested
```
Comment (tasn) [inline relay/relay.go:220]: Is there no select fall-through in Go?

**Review by tasn — APPROVED** [review]
I think it looks fine overall, though is there no "reconnecting-websocket" in Go that we can just use?

File: relay/relay.go (hunk 1 above)
Comment (svix-frank) [reply relay/relay.go:220]: There's fallthrough for switch statements but not selects

**Comment by svix-frank** [comment]
> I think it looks fine overall, though is there no "reconnecting-websocket" in Go that we can just use?

Not in the `gorilla/websocket` library which is the defacto standard.  There are some small projects but they basically just do this.

**Pull request merged**